exercises/concept/bird-watcher/.docs/hints.md

```diff
@@ -20,7 +20,8 @@
 
 ## 4. Check if there was a day with no visiting birds
 
-- The `Array` class has a built-in method that returns the first index where the element is found, or -1 if no matching element was found.
+- The `Arrays` class has a built-in method [`binarySearch`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Arrays.html) that returns the first index where the element is found, or a negative value if no matching element was found.
```

Comment:
Actually, `binarySearch` is the wrong method to use because it requires the provided array to be sorted beforehand (which is not the case here). In fact, there is no single method that directly answers the question in the `Arrays` class. Instead, you could use the `stream` method with a `fliter` and a `findFirst` then check the returned `Optional` to see if you found a match.

Personally, I think we should get rid of this line altogether since it does not actually help solve the problem simply (and we probably should not introduce streams like this).

Reply:
I used the `sort` and the `binarySearch` to solve this challenge. Is not better to use the Java API, first, if possible?


We can change the line as follow's or something similar that keeps both options open?

- If the array is sorted, the `Arrays` class has a built-in method [`binarySearch`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Arrays.html) that returns the first index where the element is found, or a negative value if no matching element was found.

Reply:
```Sort``` and ```binarySearch``` would result in  O(N LOG N) complexity due to sorting.  Searching for an element Array could be made in linear time.

Reply:
Hum, I get it. So it makes sense to avoid that approach. Will change it, thanks.